src/subgraph/index.ts

```diff
@@ -134,4 +143,59 @@ export class Subgraph {
     const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
     return getAllVaults(subgraphEndpoint);
   }
+
+  public async getChainVaultData(): Promise<Vault[]> {
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    return getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+  }
+
+  public async getTotalPoolValue() {
+    await this.init();
+    const subgraphEndpoint = this.getSubgraphEndpoint(this.rpcConfig.chainId);
+    const vaults = await getChainVaultData(this.rpcConfig.chainId, subgraphEndpoint);
+    const priceIds = vaults.map((v) => v.depositToken?.pyth?.id);
+    const res = await this.pyth.getLatestPricesFromPyth(priceIds as string[]);
+    const price = res.map((pythPrice) => {
+      const normalizedPrice = this.pyth.normalizePythPriceForParifi(
+        Number(pythPrice.price.price),
+        pythPrice.price.expo,
+      );
+      const colletralPrice = new Decimal(normalizedPrice).div(10 ** PRICE_FEED_DECIMALS);
+      const returnObj = { priceId: `0x${pythPrice.id}`, normalizedPrice: colletralPrice };
+      return returnObj;
+    });
+
+    function getNormalizedPriceById(
+      priceId: string,
+      prices: { priceId: string | undefined; normalizedPrice: Decimal }[],
+    ) {
+      // Loop through the prices array
+      for (const price of prices) {
+        // Check if the priceId matches
+        if (price.priceId == priceId) {
+          // Return the normalizedPrice if matched
+          return price.normalizedPrice;
+        }
+      }
+      // Return null if no matching priceId found
+      return null;
+    }
+
+    const data = vaults.map((vault) => {
+      const normalizedPrice = getNormalizedPriceById(vault.depositToken?.pyth?.id as string, price);
```

Comment:
if this returns null, then lets set it to 0 instead or handle that scenario better

Reply:
done